src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1053,16 +1055,71 @@ public void Connect(IPAddress[] addresses, int port)
                 throw new NotSupportedException(SR.net_invalidversion);
             }
 
+// Disable CS0162: Unreachable code detected
+//
+// SuportsMultipleConnectAttempts is a constant; when false, the following lines will trigger CS0162.
+#pragma warning disable 162
             Exception lastex = null;
-            foreach (IPAddress address in addresses)
+            if (SocketPal.SupportsMultipleConnectAttempts)
```

Comment:
> why not have this extracted within PAL?

The algorithm is not exactly platform-dependent, it's capability-dependent. All a new platform has to do for this code is declare is whether or not it supports multiple connection attempts on a socket--moving it behind the PAL would likely require a fair amount of code copying. I'd be more inclined to guard this code with `#ifdef`s than to move it behind the PAL.